torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1109,6 +1109,13 @@ def method_tests():
         ('view_as', (S, S, S), (non_differentiable(torch.rand(S * S, S)),)),
         ('view_as', (), (non_differentiable(torch.tensor(5.5)),), 'scalar'),
         ('view_as', (), (non_differentiable(torch.rand(1, 1)),), 'scalar_to_dims'),
+        ('broadcast_to', (S, 1, 1), (dont_convert((S, S, S)),), '', (False,)),
```

Comment:
We're actually starting to deprecate `method_tests()` entries like these in favor of writing OpInfos for new operators, like this:

https://github.com/pytorch/pytorch/blob/9417e92722f1b6f75e162cbe661d3ed788d8d37a/torch/testing/_internal/common_methods_invocations.py#L420

Basically, an OpInfo is added to this alphabetical list. It has to specify which dtypes it supports using the dtype functions defined here:

https://github.com/pytorch/pytorch/blob/9417e92722f1b6f75e162cbe661d3ed788d8d37a/torch/testing/__init__.py#L278

and provide the same information as these `method_tests()` entries like addmm does. For example, instead of these tuples the operator create a `sample_inputs_func` that returns them.

This is a new thing, which is a pain, but I think it will be helpful to learn and I would appreciate your feedback on it. It may be a little trickier to setup than `method_tests()` but it will make testing much easier.

One problem with OpInfos currently is that operations which support the bfloat16 datatype have to skip the following test:

```
skips=(
  # RuntimeError: "rsqrt_cuda" not implemented for 'BFloat16'
  SkipInfo(
    'TestCommon', 'test_variant_consistency_jit',
    device_type='cuda', dtypes=[torch.bfloat16])))
```

This is a bug in the test's implementation that we expect to fix soon. Please let me know if you have any questions about doing this. If creating an OpInfo is too much of a pain then we can ignore it for now and continue to use `method_tests`.

Reply:
I'm truly grateful for this comprehensive explanation of the new `OpInfo` test framework.
I think it is much more descriptive and readable compared to the `method_tests`, as the long tuple line full of testing parameters is often hard to read.
Moreover, `OpInfo` is awesome because it provides the standard way of testing newly-added operators. It helps us develop code in a more TDD style and write more robust code.

Reply:
I have rewritten the test using `OpInfo`, and updated the `test_ops.py` as `OpInfo` currently does not support testing operators whose parameters are not `torch.Tensor`.

Reply:
I think it does support non-tensor parameters, but let me take a look and I'll comment inline.